Content.Client/Parallax/Data/GeneratedParallaxTextureSource.cs

```diff
@@ -0,0 +1,143 @@
+    /// Parallax config path (the TOML file).
+    /// In client resources.
+    /// </summary>
+    [DataField("configPath")]
+    public ResourcePath ParallaxConfigPath { get; } = new("/parallax_config.toml");
+
+    /// <summary>
+    /// ID for debugging, caching, and so forth.
+    /// The empty string here is reserved for the original parallax.
+    /// It is advisible to provide a roughly unique ID for any unique config contents.
+    /// </summary>
+    [DataField("id")]
+    public string Identifier { get; } = "other";
+
+    /// <summary>
+    /// Cached path.
+    /// In user directory.
+    /// </summary>
+    private ResourcePath ParallaxCachedImagePath => new($"/parallax_{Identifier}cache.png");
+
+    /// <summary>
+    /// Old parallax config path (for checking for parallax updates).
+    /// In user directory.
+    /// </summary>
+    private ResourcePath PreviousParallaxConfigPath => new($"/parallax_{Identifier}config_old");
+
+    async Task<Texture> IParallaxTextureSource.GenerateTexture()
+    {
+        var parallaxConfig = GetParallaxConfig();
+        if (parallaxConfig == null)
+        {
+            Logger.ErrorS("parallax", "Parallax config not found or unreadable.");
+            // The show must go on.
+            return Texture.Transparent;
+        }
+
+        var debugParallax = IoCManager.Resolve<IConfigurationManager>().GetCVar(CCVars.ParallaxDebug);
+
+        if (debugParallax
+            || !StaticIoC.ResC.UserData.TryReadAllText(PreviousParallaxConfigPath, out var previousParallaxConfig)
+            || previousParallaxConfig != parallaxConfig)
+        {
+            var table = Toml.ReadString(parallaxConfig);
+            await UpdateCachedTexture(table, debugParallax);
+
+            //Update the previous config
+            using var writer = StaticIoC.ResC.UserData.OpenWriteText(PreviousParallaxConfigPath);
+            writer.Write(parallaxConfig);
+        }
+
+        try
+        {
+            return GetCachedTexture();
+        }
+        catch (Exception ex)
+        {
+            Logger.ErrorS("parallax", $"Couldn't retrieve parallax cached texture: {ex}");
+            // The show must go on.
+            try
+            {
+                // Also try to at least sort of fix this if we've been fooled by a config backup
+                StaticIoC.ResC.UserData.Delete(PreviousParallaxConfigPath);
+            }
+            catch (Exception)
+            {
+            }
+            return Texture.Transparent;
+        }
+    }
+
+    private async Task UpdateCachedTexture(TomlTable config, bool saveDebugLayers)
+    {
+        var debugImages = saveDebugLayers ? new List<Image<Rgba32>>() : null;
+
+        var sawmill = IoCManager.Resolve<ILogManager>().GetSawmill("parallax");
+        // Generate the parallax in the thread pool.
+        using var newParallexImage = await Task.Run(() =>
+            ParallaxGenerator.GenerateParallax(config, new Size(1920, 1080), sawmill, debugImages));
```

Comment:
Parallax resolution should be a user config option, maybe automatically detected from the user's display resolution.

Reply:
This is from the old code, I deliberately didn't alter it so that the old parallax would work as-is.